### ISSUE TITLE
Documentation 2.0

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/react-advanced-form#252]
## What
Suggest to consider implementing a more profound documentation.

## Why
1. Better developer experience.
2. Better newcomers guidance.
3. Better overall improvement from the solution.

## How
1. Research on service solutions.
2. Potentially develop a small doc app.
3. Live code examples are very anticipated.

**Comment by kettanaito** [comment]
As one of the readers has pointed out, it's nice to illustrate some benefits of the validation based on pure resolvers. It can cover even complex scenarios quite easily. Perhaps, having a list of various validation examples (only schemas) would be nice to have.

**Comment by kettanaito** [comment]
I must say that the updated Gitbook looks very promising. If I be able to integrate a live examples there, then the place for Docs 2.0 is defined.

**Comment by kettanaito** [comment]
Need to consider to add the comparison section to the README page.

Comparison illustrate the same scenarios achieved using different form libraries. It will help people to understand the differences.

**Comment by johnpreed** [comment]
What I find lacking is documentation on how to use field options like mapPropsToField, enforceProps, mapping/setting initial values of components, beforeRegister, shouldUpdateRecord. Detailed explanations of these along with examples would go a long way.

**Comment by kettanaito** [comment]
You are completely right, @johnpreed. I will try to take the original requirements why those methods were implemented, and put them in a form of examples.
